registry/registry.go

```diff
@@ -117,25 +155,35 @@ func (reg *registry) tagsToRepository(remote Remote, repository Repository, tags
 		images[i] = res.image
 	}
 
-	sort.Sort(byCreatedDesc(images))
+	sort.Sort(flux.ByCreatedDesc(images))
 	return images, nil
 }
 
-// -----
-
-type byCreatedDesc []flux.Image
-
-func (is byCreatedDesc) Len() int      { return len(is) }
-func (is byCreatedDesc) Swap(i, j int) { is[i], is[j] = is[j], is[i] }
-func (is byCreatedDesc) Less(i, j int) bool {
-	switch {
-	case is[i].CreatedAt.IsZero():
-		return true
-	case is[j].CreatedAt.IsZero():
-		return false
-	case is[i].CreatedAt.Equal(is[j].CreatedAt):
-		return is[i].ID.String() < is[j].ID.String()
-	default:
-		return is[i].CreatedAt.After(is[j].CreatedAt)
+// ---
+
+// This is an interface that represents the heroku docker registry library
+type HerokuRegistryLibrary interface {
+	Tags(repository string) (tags []string, err error)
+	Manifest(repository, reference string) ([]schema1.History, error)
+}
+
+// ---
+
+// Convert between types. dockerregistry returns the *same* type but from a
+// vendored library. Because golang doesn't like to apply interfaces to a
+// vendored type, we have to provide an adaptor to isolate it.
```

Comment:
I suspect this is to do with having the same type returned by two different vendored packages. Let's see what Phil says, and hence or otherwise investigate in another thread.